core/corehttp/gateway_handler.go

```diff
@@ -133,6 +133,7 @@ func (i *gatewayHandler) getOrHeadHandler(w http.ResponseWriter, r *http.Request
 	}
 
 	w.Header().Set("X-IPFS-Path", urlPath)
+	w.Header().Set("Access-Control-Allow-Origin", "*")
```

Comment:
Yes, of course - adding configuration options makes the program more
complex both to use and maintain. And in this case, being able to define
arbitrary headers moves into the territory of [softcoding](
http://thedailywtf.com/articles/Soft_Coding).

I think that adding arbitrary headers should be handled by another tool
(like nginx), because changing response headers has nothing to do with
IPFS. We should just provide reasonable defaults and let more advanced
things be handled by tools that are made to deal with them.
On Jul 7, 2015 9:04 PM, "Juan Batiz-Benet" <notifications@github.com> wrote:

> In core/corehttp/gateway_handler.go
> <https://github.com/ipfs/go-ipfs/pull/1444#discussion_r34108168>:
>
> > @@ -133,6 +133,7 @@ func (i *gatewayHandler) getOrHeadHandler(w http.ResponseWriter, r *http.Request
> >  	}
> >
> >  	w.Header().Set("X-IPFS-Path", urlPath)
> > +	w.Header().Set("Access-Control-Allow-Origin", "*")
>
> @slang800 <https://github.com/slang800> do you lose anything if we allow
> you to specify gateway headers thus:
>
> ipfs config Gateway.Headers.Access-Control-Allow-Origin "*"
> # or
> ipfs config Gateway.Headers.Access-Control-Allow-Origin "<origin>"
> # or
> ipfs config Gateway.Headers.Access-Control-Allow-Origin "mysite.com"
>
> ?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/ipfs/go-ipfs/pull/1444/files#r34108168>.
>